msal/src/main/java/com/microsoft/identity/client/internal/CommandParametersAdapter.java

```diff
@@ -586,6 +605,9 @@ public static SignInStartCommandParameters createSignInStartCommandParameters(
                 .clientId(configuration.getClientId())
                 .challengeType(configuration.getChallengeTypes())
                 .scopes(scopes)
+                // Start of the flow, so there is no correlation ID to use from a previous API response.
+                // Set it to a default value.
+                .correlationId(DiagnosticContext.INSTANCE.getThreadCorrelationId())
```

Comment:
The value returned from the method call `DiagnosticContext.INSTANCE.getThreadCorrelationId()` is always going to be the constant value `DiagnosticContext.UNSET`. We should use that value instead of making the function call.

Reply:
> The value returned from the method call DiagnosticContext.INSTANCE.getThreadCorrelationId() is always going to be the constant value DiagnosticContext.UNSET

What makes you say that?

Please have a look at usages of `setRequestContext()`. Furthermore, manual E2E tests have shown that this field is correctly set, rather than it being value `UNSET`.

Reply:
Thanks, I can see it getting updated via the `setRequestContext` method. The usage of `getThreadCorrelationId()` here is singleton pattern and it will cause issues if the developer creates multiple `NativeAuthPublicClientApplication` objects.

Reply:
That's a good point. Could you create a ticket for this on our engineering backlog? I'm not sure what the resolution should be, as this is how MSAL does it as well. But we can discuss that separately. If you agree, I'd like to leave this out of scope of this PR however.